packages/aws-cdk-lib/aws-s3-assets/lib/asset.ts

```diff
@@ -137,6 +137,10 @@ export class Asset extends Construct implements cdk.IAsset {
   constructor(scope: Construct, id: string, props: AssetProps) {
     super(scope, id);
 
+    if (!props.path) {
+      throw new Error('Asset path cannot be empty');
+    }
```

Comment:
It might also be a good idea to check that `cdk.out` is excluded from the asset source, as `Code.fromAsset('.')` would cause the same reported issue.

Reply:
+1 on this